accord-core/src/test/java/accord/topology/TopologyRandomizer.java

```diff
@@ -118,7 +119,9 @@ private static Shard[] updateBoundary(Shard[] shards, RandomSource random)
 
     private static Shard[] split(Shard[] shards, RandomSource random)
     {
-        int idx = random.nextInt(shards.length - 1);
+        if (shards.length == 0)
+            throw new IllegalArgumentException("Unable to split an empty array");
+        int idx = shards.length == 1 ? 0 : random.nextInt(shards.length - 1);
```

Comment:
now that topology randomizer is not no-oping again, this following case was seen

T0 merge
T1 merge
T2 merge -- now single shard
T3 split - failed